src/briefcase/platforms/linux/system.py

```diff
@@ -994,6 +1018,115 @@ def _package_rpm(self, app: AppConfig, **kwargs):
+                "App configuration does not define `description`. "
+                "Arch projects require a description."
+            )
+        # The changelog should exist.
+        changelog_source = self.base_path / "CHANGELOG"
+        if not changelog_source.exists():
+            raise BriefcaseCommandError(
+                """\
+Your project does not contain a CHANGELOG file.
+Create a file named `CHANGELOG` in the same directory as your `pyproject.toml`
+with details about the release.
+"""
+            )
+            
+        # Generate the pkgbuild layout
+        pkgbuild_path = self.build_path(app) / "pkgbuild"
+        with self.input.wait_bar("Generating pkgbuild layout..."):
+            if pkgbuild_path.exists():
+                self.tools.shutil.rmtree(pkgbuild_path)
+            (pkgbuild_path).mkdir(parents=True)
+
+            # Build the source archive
+            with self.input.wait_bar("Building source archive..."):
+                
+                # Copy the CHANGELOG file to the bundle_path so it can be included in the source archive
+                self.tools.shutil.copy(changelog_source, self.bundle_path(app) / "CHANGELOG")
+                
+                self.tools.shutil.make_archive(
+                    pkgbuild_path / f"{app.app_name}-{app.version}",
+                    format="gztar",
+                    root_dir=self.build_path(app),
+                    base_dir=self.bundle_path(app),
+                )
+            # Write the arch PKGBUILD file.
+
+            # Add runtime package dependencies. App config has been finalized,
+            # so this will be the target-specific definition, if one exists.
+            system_runtime_requires_list = [
+                f"glibc>={app.glibc_version}",
+                f"python{app.python_version_tag}",
+                "gobject-introspection",
+                "gobject-introspection-runtime",
+                "gtk3",
+            ] + getattr(app, "system_runtime_requires", [])
+
+            system_runtime_requires = " ".join(
+                f"'{pkg}'" for pkg in system_runtime_requires_list
+            )
+
+            with (pkgbuild_path / "PKGBUILD").open("w", encoding="utf-8") as f:
+                f.write(
+                    "\n".join(
+                        [
+                            f"# Maintainer: {app.author} <{app.author_email}>",
+                            f"pkgname={app.app_name}",
+                            f"pkgver={app.version}",
+                            f"pkgrel={getattr(app, 'revision', 1)}",
+                            f'pkgdesc="{app.description}"',
+                            f"arch=('any')",  # We should use {self.linux_arch}, but doing this as the architecture error 
+                                              # was probably caused due to this parameter.
+                                              # Moreover arch wiki recommends 'any' for architecture independent packages.
```

Comment:
This doesn't smell right to me. 

If you have `numpy` (or any other binary dependency) in your app, Briefcase will be installing an architecture-specific binary wheel. Python is able to guarantee that the wheel will work because it adheres to the `manylinux` specification, which constraints the libraries that the wheel can link against; but it means that the artefacts in the wheel are platform specific. If you build your package on x86-64, you won't be able to deploy it on arm64.

Reply:
The architecture error is usually caused if the host machine and target package architectures are different. That means, the previous error was caused due to mismatch of host architecture string. I have modified it to use `self.tools.host_arch`. Now, it should work fine.